### PR TITLE
Filter phone# spaces in SignUp process for issue# 33

### PR DESCRIPTION
**Pull request by remivar**
Created UpdateUsername() function in the SignInViewModel. The function will be called from the SignInViewController, which in turn calls Validation.cleanedPhoneNumber to filter phone# by removing space , +,-,(,), etc.

### CODE REVIEW

File: TreeTracker/Services/User/LoginService.swift [hunk 1]
```diff
@@ -31,6 +31,9 @@ class LocalLoginService: LoginService {
 
     func login(withUsername username: Username, completion: (Result<Planter, Error>) -> Void) {
 
+
```
Comment (AlexCornforth) [inline TreeTracker/Services/User/LoginService.swift]: Guessing this was left in by mistake?

File: TreeTracker/UI/Views/AddTree/AddTree.storyboard [hunk 2]
```diff
@@ -1,12 +1,11 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<document type="com.apple.InterfaceBuilder3.CocoaTouch.Storyboard.XIB" version="3.0" toolsVersion="17701" targetRuntime="iOS.CocoaTouch" propertyAccessControl="none" useAutolayout="YES" useTraitCollections="YES" useSafeAreas="YES" colorMatched="YES" initialViewController="QqA-0g-MHc">
```
Comment (AlexCornforth) [inline TreeTracker/UI/Views/AddTree/AddTree.storyboard]: Please avoid committing code unrelated to the change. Would be great if you could revert these storyboard changes back.

File: TreeTracker/UI/Views/Authentication/SignIn/SignInViewModel.swift [hunk 3]
```diff
@@ -41,7 +41,7 @@ class SignInViewModel {
 
     let title: String = L10n.SignIn.title
 
-    private var loginType: LoginType = .phoneNumber {
```
Comment (AlexCornforth) [inline TreeTracker/UI/Views/Authentication/SignIn/SignInViewModel.swift]: I think we want to keep this private - its only used inside the viewModel

File: TreeTracker/UI/Views/Authentication/SignIn/SignInViewModel.swift [hunk 4]
```diff
@@ -53,15 +53,19 @@ class SignInViewModel {
         self.loginType = loginType
     }
 
-    private var usernameValue: String = "" {
+    var usernameValue: String = "" {
```
Comment (AlexCornforth) [inline TreeTracker/UI/Views/Authentication/SignIn/SignInViewModel.swift:56]: I think we want to keep this private - its only used inside the viewModel

File: TreeTracker/UI/Views/Authentication/SignIn/SignInViewModel.swift [hunk 5]
```diff
@@ -53,15 +53,19 @@ class SignInViewModel {
...
         didSet {
             viewDelegate?.signInViewModel(self, didUpdateValidationState: usernameValid)
             viewDelegate?.signInViewModel(self, didUpdateLoginEnabled: loginEnabled)
         }
     }
 
     func updateUsername(username: String) {
-        usernameValue = username
+        if loginType == .phoneNumber {
+            usernameValue = Validation.cleanedPhoneNumber(phoneNumber: username) }
```
Comment (AlexCornforth) [inline TreeTracker/UI/Views/Authentication/SignIn/SignInViewModel.swift]: I still think we should only gtrim the whitepsace here  rather than whitespace and `+`, `-`, `(`, `)`. The area code including the + is a valid part of the phone number. This issue we were seeing here is when iOS auto detects phone numbers it can add in some whitespace in pasting.

The android app does a trim here on both phone and email clearing any trailing & leading whitespace, i think we should still clear all whitespace though. See the [LoginViewModel.kt](https://github.com/Greenstand/treetracker-android/blob/4b8d9dbf186b9933ba66716c48f405ae01a9456f/app/src/main/java/org/greenstand/android/TreeTracker/viewmodels/LoginViewModel.kt)

**Review by AlexCornforth — CHANGES_REQUESTED** [review]
I think we should only trim whitespace rather than the other characters - this is closer to what the android app does too.

Also please can you revert the storyboard changes too so the pull request only contains code relevant to the ticket.

File: TreeTracker/Services/User/LoginService.swift (hunk 1 above)
Comment (remivar) [reply TreeTracker/Services/User/LoginService.swift]: am sorry, I removed the comment line that has my name by mistake

File: TreeTracker/UI/Views/AddTree/AddTree.storyboard (hunk 2 above)
Comment (remivar) [reply TreeTracker/UI/Views/AddTree/AddTree.storyboard]: The XML file in my branch is 

<?xml version="1.0" encoding="UTF-8"?>
<document type="com.apple.InterfaceBuilder3.CocoaTouch.Storyboard.XIB" version="3.0" toolsVersion="16097.2" targetRuntime="iOS.CocoaTouch" propertyAccessControl="none" useAutolayout="YES" useTraitCollections="YES" useSafeAreas="YES" colorMatched="YES" initialViewController="QqA-0g-MHc">

Should I change the value of  toolsVersion?

Screen Shot 2021-06-09 at 1 10 42 PM

File: TreeTracker/UI/Views/Authentication/SignIn/SignInViewModel.swift (hunk 3 above)
Comment (remivar) [reply TreeTracker/UI/Views/Authentication/SignIn/SignInViewModel.swift]: Okay. I changed it as private

File: TreeTracker/UI/Views/Authentication/SignIn/SignInViewModel.swift (hunk 4 above)
Comment (remivar) [reply TreeTracker/UI/Views/Authentication/SignIn/SignInViewModel.swift:56]: It caused error in the Validation.swift. Therefore I had to make it as public property.

File: TreeTracker/UI/Views/Authentication/SignIn/SignInViewModel.swift (hunk 5 above)
Comment (remivar) [reply TreeTracker/UI/Views/Authentication/SignIn/SignInViewModel.swift]: I deleted the codes that remove the occurrences of +,-(). However, when we run the build the simulator doesn't enable the Signin button if the text field has an input that includes special characters.

File: TreeTracker/Utilities/Validation.swift [hunk 6]
```diff
@@ -75,12 +81,4 @@ private extension Validation {
         return false
     }
 
-    static func cleanedPhoneNumber(phoneNumber: String) -> String {
```
Comment (AlexCornforth) [inline TreeTracker/Utilities/Validation.swift:78]: Please can you revert this change. It could cause invalid phone numbers to be entered.

We can just apply `.replacingOccurrences(of: " ", with: "")` in the view model for this change

File: TreeTracker/UI/Views/Authentication/SignIn/SignInViewModel.swift [hunk 7]
```diff
@@ -53,15 +53,19 @@ class SignInViewModel {
...
+    var usernameValue: String = "" {
         didSet {
             viewDelegate?.signInViewModel(self, didUpdateValidationState: usernameValid)
             viewDelegate?.signInViewModel(self, didUpdateLoginEnabled: loginEnabled)
         }
     }
 
     func updateUsername(username: String) {
-        usernameValue = username
+        if loginType == .phoneNumber {
```
Comment (AlexCornforth) [inline TreeTracker/UI/Views/Authentication/SignIn/SignInViewModel.swift]: Maybe a switch on login type here would be better?

**Review by AlexCornforth — CHANGES_REQUESTED** [review]

File: TreeTracker/UI/Views/AddTree/AddTree.storyboard (hunk 2 above)
Comment (remivar) [reply TreeTracker/UI/Views/AddTree/AddTree.storyboard]: I resolved the conflicts in AddTree.storyboard

File: TreeTracker/UI/Views/Authentication/SignIn/SignInViewModel.swift (hunk 7 above)
Comment (remivar) [reply TreeTracker/UI/Views/Authentication/SignIn/SignInViewModel.swift]: I used switch case here. 

switch loginType {
        case .phoneNumber:
            usernameValue = Validation.cleanedPhoneNumber(phoneNumber: username)
        case .email:
            usernameValue = username
        }

File: TreeTracker/Utilities/Validation.swift (hunk 6 above)
Comment (remivar) [reply TreeTracker/Utilities/Validation.swift:78]: I added the below codes back to the func cleanedPhoneNumber()
             .replacingOccurrences(of: "+", with: "")
             .replacingOccurrences(of: " ", with: "")
             .replacingOccurrences(of: "(", with: "")
             .replacingOccurrences(of: ")", with: "")
             .replacingOccurrences(of: "-", with: "")

Comment (AlexCornforth) [reply TreeTracker/Utilities/Validation.swift:78]: Please could you also move it back into the private extension

File: TreeTracker/UI/Views/Authentication/SignIn/SignInViewModel.swift [hunk 8]
```diff
@@ -53,15 +53,20 @@ class SignInViewModel {
...
-    private var usernameValue: String = "" {
+    var usernameValue: String = "" {
         didSet {
             viewDelegate?.signInViewModel(self, didUpdateValidationState: usernameValid)
             viewDelegate?.signInViewModel(self, didUpdateLoginEnabled: loginEnabled)
         }
     }
 
     func updateUsername(username: String) {
-        usernameValue = username
```
Comment (AlexCornforth) [inline TreeTracker/UI/Views/Authentication/SignIn/SignInViewModel.swift:64]:
```
func updateUsername(username: String) {
    usernameValue = username.replacingOccurrences(of: "+", with: "")
}
```
Will do the job here. Ok to trim e ails too as they shouldnt have spaces.

**Comment by AlexCornforth** [comment]
Going to close this as now fixed in [this pr](https://github.com/Greenstand/treetracker-ios/pull/70).
